aea/helpers/profiling.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Implementation of background profiling daemon."""
+
+import asyncio
+import datetime
+import gc
+import logging
+import platform
+import textwrap
+import threading
+import time
+from collections import Counter
+from typing import Callable
+
+from aea.connections.base import Connection
+from aea.helpers.async_utils import Runnable
+from aea.protocols.base import Message
+from aea.protocols.dialogue.base import Dialogue
+from aea.skills.base import Behaviour, Handler
```

Comment:
This heavily relies on aea modules. I'd move the Profiling class to its own `aea` subdirectory and only keep the non-aea specific methods here

Reply:
refactored